Application Directory Structure

yugandhargangu opened the pull request:
1. Bettle track application directory structure.
2. Power point file of directory
3. Gradle script files and sub modules.
4. Spring boot application configuration

yugandhargangu committed:
Application Directory Structure

1. Bettle track application directory structure.
2. Power point file of directory
3. Gradle script files and sub modules.
4. Spring boot application configuration

yugandhargangu committed:
Update .travis.yml

yugandhargangu committed:
Update .travis.yml

yugandhargangu committed:
Resolved build failure issue.

yugandhargangu committed:
Build fail issue resolved.

yugandhargangu committed:
Build fail issue resolved.

yugandhargangu committed:
Build failed issue patch.

yugandhargangu committed:
Build failure resolved.

shashankitmaster commented on .travis.yml:
```diff
@@ -0,0 +1,12 @@
+
+jdk:
+
+  - oraclejdk8
```
Please use openjdk for java instead of oracle.

yugandhargangu replied:
travis-ci doesn't have openjdk8.
https://github.com/travis-ci/travis-ci/issues/6483

shashankitmaster replied:
I got your point but I think Oracle will soon licence JDK use.

shashankitmaster commented on application/bettle/.project:
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Remove this file and above `.classpath` . add `.gitignore` file

yugandhargangu replied:
Done.

shashankitmaster commented on application/bettle/bettle-entity/build.gradle:
```diff
@@ -0,0 +1,9 @@
+apply plugin: "java"
```
@ranjeet1515 : Add lombok plugin and reduce the code of setter and getter to annotation. 

Refer: https://projectlombok.org/mavenrepo/

yugandhargangu replied:
Done

shashankitmaster commented on application/bettle/bettle-jpa/src/main/java/com/bettle/track/jpa/repos/BaseRepository.java:
```diff
@@ -0,0 +1,62 @@
+public interface BaseRepository<T extends AbstractParentEntity> extends CrudRepository<T, Long>, JpaRepository<T, Long> {
+
+    @Override
+    @Query("UPDATE #{#entityName} e SET e.active_flag = 0 WHERE e.id = :id")
```
Can we move queries to resources folder and make use of annotation with fetches the query from classpath.

yugandhargangu replied:
Yes. This part will be done in jpa specific branch but not in this branch. So I removed this class.

shashankitmaster replied:
@yugandhargangu : Got it

shashankitmaster commented:
Change destination branch to `release` branch

yugandhargangu committed:
Review changes

yugandhargangu committed:
Updated with review changes.

shashankitmaster commented on application/bettle/bettle-common/src/main/java/com/bettle/track/enums/CustomFieldType.java:
```diff
@@ -0,0 +1,57 @@
+        this.required = required;
+    }
+
+    public int getId() {
```
@yugandhargangu : Please get rid of getter and setter as we using lombok plugin for that.
 
Please try to do that for all classes

shashankitmaster replied:
Okay. I am sorry I thought those are classes. I am approving.

yugandhargangu committed:
Added two modules

1. bettle-repo module
   - Contains all repository interfaces
2. bettle-service module
   - COntains all service classes

yugandhargangu commented:
Updated with bettle-repo and bettle-service modules.

shashankitmaster reviewed (approved):
Please merge ur branch
